l1-contracts/contracts/bridgehub/Bridgehub.sol

```diff
@@ -269,27 +281,53 @@ contract Bridgehub is IBridgehub, ReentrancyGuard, Ownable2StepUpgradeable, Paus
         baseTokenAssetId[_chainId] = DataEncoding.encodeNTVAssetId(block.chainid, _baseToken);
         settlementLayer[_chainId] = block.chainid;
 
-        IStateTransitionManager(_stateTransitionManager).createNewChain({
+        address chainAddress = IStateTransitionManager(_stateTransitionManager).createNewChain({
             _chainId: _chainId,
             _baseToken: _baseToken,
             _sharedBridge: address(sharedBridge),
             _admin: _admin,
             _initData: _initData,
             _factoryDeps: _factoryDeps
         });
+        _registerNewHyperchain(_chainId, chainAddress);
         messageRoot.addNewChain(_chainId);
 
         emit NewChain(_chainId, _stateTransitionManager, _admin);
         return _chainId;
     }
 
+    /// @dev This internal function is used to register a new hyperchain in the system.
+    function _registerNewHyperchain(uint256 _chainId, address _hyperchain) internal {
+        // slither-disable-next-line unused-return
+        hyperchainMap.set(_chainId, _hyperchain);
+        require(hyperchainMap.length() <= MAX_NUMBER_OF_HYPERCHAINS, "STM: Hyperchain limit reached");
+    }
+
     /*//////////////////////////////////////////////////////////////
                              Getters
     //////////////////////////////////////////////////////////////*/
 
-    /// @notice return the state transition chain contract for a chainId
-    function getHyperchain(uint256 _chainId) public view returns (address) {
-        return IStateTransitionManager(stateTransitionManager[_chainId]).getHyperchain(_chainId);
+    /// @notice Returns all the registered hyperchain addresses
+    function getAllHyperchains() public view override returns (address[] memory chainAddresses) {
+        uint256[] memory keys = hyperchainMap.keys();
+        chainAddresses = new address[](keys.length);
+        uint256 keysLength = keys.length;
+        for (uint256 i = 0; i < keysLength; ++i) {
+            chainAddresses[i] = hyperchainMap.get(keys[i]);
+        }
+    }
+
+    /// @notice Returns all the registered hyperchain chainIDs
+    function getAllHyperchainChainIDs() public view override returns (uint256[] memory) {
```

Comment:
BTW, maybe move this mapping to a separate contract (that will be fully controlled by bridgehub)? STM is at breaking capacity wrt to contracts' size now. Let's prevent this situation from bridgehub by including the getters methods elsewhere. 

We can also move the logic for bridging there also. 

Though all this can be done as a separate pr

Reply:
What is breaking capacity? We can go up to 24k, STM is 18, BH is 15.

Reply:
Okay, probably we've already moved enough logic out of STM (esp wrt the genesis upgrade). When I first added new functionality there I was constantly bumping into the 24kb limit for STM.

If rn it is 16.5/24 for BH I don't see any need to break it up indeed

Reply:
<img width="695" alt="image" src="https://github.com/user-attachments/assets/74c63389-d334-4099-a464-29cfbee11820">